src/main/java/org/folio/innreach/mapper/InnReachTransactionMapper.java

```diff
@@ -59,4 +86,173 @@ public InnReachTransactionsDTO toDTOCollection(Page<InnReachTransaction> pageabl
 
     return new InnReachTransactionsDTO().transactions(dtos).totalRecords((int) pageable.getTotalElements());
   }
+
+  public InnReachTransaction toEntity(InnReachTransactionDTO transactionDTO) {
```

Comment:
hmm.. the majority of fields can be mapped automatically by MapStruct. there are also tools to customize mapping. Please read the documentation and optimize the code

Reply:
optimized